### DISCUSSION TITLE
Collect the number of results returned, the KQL and the user

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [microsoft-search/pnp-modern-search#3136] [Q&A]
I was asked to log in a SharePoint list, all the Queries users asked for and the number of results they got.

I have a custom page called with ?k={ kql with our metadata} that includes PnP-search results and the Search box and the filters. 
I want to be able to add an item with the value of k, the number of results and the user ID.
Is there a way I can do this with the search-extensibility? 
I can easily make an other web part that will get the query string and the user. 
How will I get the results from the  PnP-search results web part?
Should I just look for `<label class="ms-fontWeight-semibold"><b>41615</b> results</label>` and read the number inside the `<b>`?

Thanks

**Comment by kasperbolarsen** [comment]
Just a verifying question: have you checked if the search log can provide the answers you are looking for?

**Comment by wobba** [comment]
@Ofer-Gal I won't comment on the privacy implication of such logging, but yes, you can use extensibility to achieve this. You could create a web component which does not render anything, and pass in the information you want to log to your SharePoint list.

**Reply by Ofer-Gal** [reply]
The search page I am customizing has nothing private. all work. users complain but never give details so we want to find out what the issues are.
Do you have an example how to get the number of results?

**Reply by kasperbolarsen** [reply]
I think Mikael is referring to the fact that this kind of logging might require formal acceptance from the users in some countries , think GDPR.

**Reply by wobba** [reply]
@Ofer-Gal in many parts of the world like where I live, you cannot log anything about what I do at work - just saying 🙂 Even at work privacy has merit for the individual. That could be under GDPR as @kasperbolarsen says, or long other privacy laws and regulations.
